homework/grayscale-image/compression.cpp

```diff
@@ -0,0 +1,54 @@
+#include "compression.hpp"
+
+#include <algorithm>
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, width>, height>& bitmap) {
+    std::vector<std::pair<uint8_t, uint8_t>> compressedVector;
+    compressedVector.reserve(width * height);
+    auto nextCharacter = ' ';
+
+    std::for_each(bitmap.cbegin(), bitmap.cend(), [&](const auto& line) {
+        std::for_each(line.cbegin(), line.cend(), [&](const auto character) {
+            if(character != nextCharacter) {
+                compressedVector.push_back(std::make_pair(character, 1));
+            }
+            else {
+                compressedVector[compressedVector.size() - 1].second += 1;
+            }
+            nextCharacter = character;
+        });
+        nextCharacter = ' ';
+    });
+
+    return compressedVector;
+}
+
+std::array<std::array<uint8_t, width>, height> decompressGrayscale(const std::vector<std::pair<uint8_t, uint8_t>>& compressedVector) {
+    std::array<std::array<uint8_t, width>, height> decompresedArray;
+    auto it = decompresedArray.begin();
+    auto positionInArray = 0;
+
+    std::for_each(compressedVector.cbegin(), compressedVector.cend(), [&](const auto& element) mutable {
```

Comment:
one std::fil_n is enough. The array is continuous in memory, so std::fill_n will automatically jump to the next line.